crates/ra_ide/src/syntax_highlighting.rs

```diff
@@ -16,6 +16,32 @@ use crate::{
     FileId,
 };
 
+const HIGHLIGHT_TAG_FIELD: &'static str = "field";
+const HIGHLIGHT_TAG_FUNCTION: &'static str = "function";
+const HIGHLIGHT_TAG_MODULE: &'static str = "module";
+const HIGHLIGHT_TAG_TYPE: &'static str = "type";
+const HIGHLIGHT_TAG_CONSTANT: &'static str = "constant";
+const HIGHLIGHT_TAG_MACRO: &'static str = "macro";
+const HIGHLIGHT_TAG_VARIABLE: &'static str = "variable";
+const HIGHLIGHT_TAG_VARIABLE_MUT: &'static str = "variable.mut";
+const HIGHLIGHT_TAG_TEXT: &'static str = "text";
+
+const HIGHLIGHT_TAG_TYPE_BUILTIN: &'static str = "type.builtin";
+const HIGHLIGHT_TAG_TYPE_SELF: &'static str = "type.self";
+const HIGHLIGHT_TAG_TYPE_PARAM: &'static str = "type.param";
+const HIGHLIGHT_TAG_TYPE_LIFETIME: &'static str = "type.lifetime";
+
+const HIGHLIGHT_TAG_LITERAL_BYTE: &'static str = "literal.byte";
+const HIGHLIGHT_TAG_LITERAL_NUMERIC: &'static str = "literal.numeric";
+const HIGHLIGHT_TAG_LITERAL_CHAR: &'static str = "literal.char";
+const HIGHLIGHT_TAG_LITERAL_COMMENT: &'static str = "comment";
+const HIGHLIGHT_TAG_LITERAL_STRING: &'static str = "string";
+const HIGHLIGHT_TAG_LITERAL_ATTRIBUTE: &'static str = "attribute";
+
+const HIGHLIGHT_TAG_KEYWORD_UNSAFE: &'static str = "keyword.unsafe";
+const HIGHLIGHT_TAG_KEYWORD_CONTROL: &'static str = "keyword.control";
+const HIGHLIGHT_TAG_KEYWORD: &'static str = "keyword";
```

Comment:
Instead of repeating `HIGHLIGHT_TAG`, let's use

```
pub mod tags {
  const TEXT: &str = "text"
}
```

Otherwise, LGTM!

bors delegate+